### ISSUE TITLE
0 odometer value captured in plugin for android

### ISSUE THREAD

**Issue by production-na-basware** [transistorsoft/cordova-background-geolocation-lt#1333]
## PROD
* Plugin version: 4.4.0
* Platform: iOS and Android
* OS version: windows
* Device manufacturer / model: Samsung, Iphone 8
* Cordova version (`cordova -v`):  11.0.0
* Cordova platform version (`cordova platform ls`): "cordova-android": "^10.1.1", and "cordova-ios": "6.1.1"
* Plugin config provided to `#ready`:
```javascript <!-- syntax-highlighting:  paste your code below -->
{
      reset: true,
      debug: environment.production ? false : true,
      logLevel: environment.production ? BackgroundGeolocation.LOG_LEVEL_ERROR : BackgroundGeolocation.LOG_LEVEL_VERBOSE,
      locationTemplate: '{"coords":{"lat":<%= latitude %>,"lng":<%= longitude %>},"event":"<%= event %>","isMoving":<%= is_moving %>,"timestamp": "<%= timestamp %>","odometer":<%= odometer %>}',
      distanceFilter: 10,
      stopTimeout: 15,
      heartbeatInterval: 5,
      foregroundService: true,
      useSignificantChangesOnly: false,
      stopOnTerminate: false,
      startOnBoot: true,
      url: vapiEndPoint,
      autoSync: false,
      maxDaysToPersist: 1,
     desiredAccuracy: BackgroundGeolocation.DESIRED_ACCURACY_HIGH,
      desiredOdometerAccuracy: 10,
      locationsOrderDirection: 'ASC',
      showsBackgroundLocationIndicator: true,
      locationAuthorizationRequest: 'Always',
      backgroundPermissionRationale: {
        title: this.prompts.lblPermTitle,
        message: this.plt.is('android') ? this.prompts.lblPermMessage : this.prompts.lblIosPermMessage,
        positiveAction: this.plt.is('android') ? this.prompts.btnPermPositiveAction : this.prompts.btnIosPermPositiveAction,
        negativeAction: this.prompts.btnPermNegativeAction
      },
      locationAuthorizationAlert: {
        titleWhenNotEnabled: this.prompts.lblTitleWhenNotEnabled,
        titleWhenOff: this.prompts.lblTitleWhenOff,
        instructions: this.plt.is('android') ? this.prompts.lblInstructions : this.prompts.lblIosInstructions,
        cancelButton: this.prompts.btnCancelButton,
        settingsButton: this.prompts.btnSettingsButton
      },
      notification: {
        text: this.prompts.lblNotificationText,
        priority: BackgroundGeolocation.NOTIFICATION_PRIORITY_DEFAULT
      },
      params: {},
      extras: {}
    }
```

## Expected Behavior
The plugin should track with complete accuracy with exact miles traveled.

## Actual Behavior
First case:
In short trips for 0.7mile to 1/1.5 mile sometimes I get 0 odometer value captured in plugin for android only.

Second case:
I traveled with a distance of 21.47miles with the plugin and The odometer on my van showed 24.9miles, this issue comes on both android and iOS. On Android, it has more discrepancy.

## Steps to Reproduce
1. I plugged the config in and added it to the app
2. when started the trip and ended it shows wrong details

## Context
I am expecting the correct odometer details with correct values.

**Comment by christocracy** [comment]
>desiredOdometerAccuracy: 10,

This means that odometer updates will occur **only** when a location has `location.accuracy <= 10` meters.  If location is `> 10` meters,  the plugin will refused to update the odometer.

**Comment by christocracy** [comment]
Note that error is introduced with each location update.  For example, imagine a trip of 10km where all locations have accuracy `5` meters.  If there were 300 locations recorded during that trip, the error would be *300 * 5 meters = 1500 meters*.

**Comment by christocracy** [comment]
There is no magic to the odometer.  The plugin simply keeps track of the last location and calculates the distance between current location and last location.

`odometer += currentLocation.distanceTo(lastLocation)`

**Comment by production-na-basware** [comment]
So shall I use the max value like desiredOdometerAccuracy = 500 or anything else?
I just want to get a correct value of which my car has traveled

**Comment by christocracy** [comment]
allowing a location with a location.accuracy of up to `500` meters would introduce undesirable error into the odometer calculation.  The default is `100`.

>I just want to get a correct value of which my car has traveled

There will always be error.  It will never match a car's odometer.

**Comment by production-na-basware** [comment]
ok, If I use value 100 then can you please explain what percentage I can get in errors for a long trip (>100km) and a short trip(<100km). Do you have any documentation for this so I can convey it to my client?

**Comment by christocracy** [comment]
>Do you have any documentation for this so I can convey it to my client?

I do not.  Each received location an [Location.coords.accuracy](https://transistorsoft.github.io/cordova-background-geolocation-lt/interfaces/coords.html#accuracy) in meters.

A geolocation is not a simple point in space.  It is a *circle* with radius *accuracy*.  The device's actual position is somewhere within that circle.

The best accuracy you'll ever see from a mobile device is *5 meters*.   If you receive a location while indoors, where location comes from *Wifi*, the accuracy is typically ~40 meters, meaning a device could be within the house beside or the house across the street.

Each recorded location introduces error of `location.coords.accuracy` into the odometer reading.

**Comment by christocracy** [comment]
As mentioned earlier, is the summation of the distance from `currentLocation` to `lastLocation`.

`odometer += currentLocation.distanceTo(lastLocation)`

Each calculation can introduce error based upon the `location.coords.accuracy` of each of the two locations the calculation.

<img width="764" alt="Screen Shot 2022-02-22 at 10 36 57 AM" src="https://user-images.githubusercontent.com/14999/155166637-079d0716-499b-41d6-84f2-8d9659cd0000.png">

**Comment by production-na-basware** [comment]
Very well explained thanks a lot for the diagram explanation. I will try this and share the result and will close the ticket.

**Comment by production-na-basware** [comment]
Fixed
